reVX/plexos/simple_plant_builder.py

```diff
@@ -254,6 +263,24 @@ def check_valid_buildouts(self, plant_sc_builds):
                 logger.error(msg)
                 raise RuntimeError(msg)
 
+            gid_caps = single_plant_sc['gid_capacity'].values.tolist()
+            gid_caps = [g for subset in gid_caps for g in subset]
+            assert len(gid_caps) == len(sc_res_gids)
+
+            plant_res_gids = [gid for i, gid in enumerate(sc_res_gids)
+                              if gid_caps[i] > 0]
+
+            shared = [gid for gid in plant_res_gids
+                      if gid in global_built_res_gids]
+
+            if any(shared) and not self._share_res:

```

Comment:
Can this happen? Do we have a negative test for this? Or is this just a triple check?

Reply:
I built in this check first as more of full-scale on-the-fly check as opposed to the pytests. This method is useful to raise any errors before we start running parallel compute. But no this should never happen, I built it in first and made sure that it would raise a good error then built the feature. It's hard to do a negative test because if you tell the object to not share resource it performs property and won't raise an error. I did add a test to make sure that it won't share resource though.

Reply:
Gotcha, I'm honestly torn, its a good check but if its never going to run I wonder if its needed? Honestly partly thinking of our code coverage score ;-), but also its fairly complex and if its not needed I'm okay removing it and relying on the pytest and your super simple elegent solution!